types/react-relay/test/relay-experimental-tests.tsx

```diff
@@ -880,3 +881,49 @@ function Subscription() {
         return <p>The subscription has been started in the background.</p>;
     };
 }
+
+/**
+ * Tests for useQueryLoader
+ * see https://relay.dev/docs/en/experimental/api-reference#usequeryloader
+ */
+function QueryLoader() {
+    const query = graphql`
+        query AppQuery($id: ID!) {
+            user(id: $id) {
+                name
+            }
+        }
+    `;
+
+    const [queryReference, loadQuery, disposeQuery] = useQueryLoader<AppQuery>(query);
+    type queryReferenceType = typeof queryReference;
+
+    function QueryFetcherExample(): React.ReactElement {
+        React.useEffect(() => {
+            loadQuery({ id: 'EXAMPLE' });
+            return disposeQuery;
+        });
+
+        return (
+            <>
+                {queryReference != null && (
+                    <>
+                        <React.Suspense fallback="Loading">
+                            <NameDisplay queryReference={queryReference} />
+                        </React.Suspense>
+                    </>
+                )}
+            </>
+        );
+    }
+
+    function NameDisplay({ queryReference }: { queryReference: NonNullable<queryReferenceType> }) {
```

Comment:
I.e. here I would expect the `queryReference` parameter to use a typing emitted by relay-compiler and the TS plugin.